static/app/views/settings/account/accountIdentities.tsx

```diff
@@ -34,54 +43,146 @@ class AccountIdentities extends AsyncView<Props, State> {
+              label={
+                identity.category === UserIdentityCategory.SOCIAL_IDENTITY
+                  ? t('App Integration')
+                  : identity.category === UserIdentityCategory.GLOBAL_IDENTITY
+                  ? t('Sign-In Identity')
+                  : identity.category === UserIdentityCategory.ORG_IDENTITY
+                  ? t('Organization Identity')
+                  : ''
+              }
+            />
+            {identity.organization && (
+              <Tag isOrgName label={identity.organization.slug} />
+            )}
+          </TagWrapper>
+
+          {this.renderButton(identity)}
+        </InternalContainer>
+      </IdentityPanelItem>
     );
   };
 
+  renderButton(identity: UserIdentityConfig) {
+    return identity.status === UserIdentityStatus.CAN_DISCONNECT ? (
+      <Confirm
+        onConfirm={() => this.handleDisconnect(identity)}
+        priority="danger"
+        confirmText={t('Disconnect')}
+        message={
+          <React.Fragment>
+            <Alert type="error" icon={<IconFlag size="md" />}>
+              {t(`Disconnect Your ${identity.provider.name} Identity?`)}
+            </Alert>
+            <TextBlock>
+              {identity.category === UserIdentityCategory.SOCIAL_IDENTITY
+                ? t("This action can't be undone.")
+                : t(
+                    'After disconnecting, you will need to use a password or another identity to sign in.'
+                  )}
+            </TextBlock>
+          </React.Fragment>
+        }
+      >
+        <Button>{t('Disconnect')}</Button>
+      </Confirm>
+    ) : (
+      <Button
+        disabled
+        title={
+          identity.status === UserIdentityStatus.NEEDED_FOR_GLOBAL_AUTH
+            ? t(
+                'You need this identity to sign into your account. If you want to disconnect it, set a password first.'
+              )
+            : identity.status === UserIdentityStatus.NEEDED_FOR_ORG_AUTH
+            ? t('You need this identity to access your organization.')
+            : null
+        }
+      >
+        {t('Disconnect')}
+      </Button>
+    );
+  }
+
+  handleDisconnect = (identity: UserIdentityConfig) => {
+    disconnectIdentity(identity, () => this.reloadData());
+  };
+
+  itemOrder = (a: UserIdentityConfig, b: UserIdentityConfig) => {
+    function categoryRank(c: UserIdentityConfig) {
+      return [
+        UserIdentityCategory.GLOBAL_IDENTITY,
+        UserIdentityCategory.SOCIAL_IDENTITY,
+        UserIdentityCategory.ORG_IDENTITY,
+      ].indexOf(c.category);
+    }
+
+    if (a.provider.name !== b.provider.name) {
+      return a.provider.name < b.provider.name ? -1 : 1;
+    } else if (a.category !== b.category) {
+      return categoryRank(a) - categoryRank(b);
+    } else if ((a.organization?.name ?? '') !== (b.organization?.name ?? '')) {
+      return (a.organization?.name ?? '') < (b.organization?.name ?? '') ? -1 : 1;
+    } else return 0;
```

Comment:
these can alll be early return.

eg, you don't need the `else`


```tsx
    if (a.provider.name !== b.provider.name) {
      return a.provider.name < b.provider.name ? -1 : 1;
    }

    if (a.category !== b.category) {
      return categoryRank(a) - categoryRank(b);
    }
    
    if ((a.organization?.name ?? '') !== (b.organization?.name ?? '')) {
      return (a.organization?.name ?? '') < (b.organization?.name ?? '') ? -1 : 1;
    }

    return 0;
```

Reply:
Just curious, is there a style guide for this? I included the `else`s so that the linter would bring the lines together and save vertical space, which some styles prefer.

I don't have a strong preference though, so I'll change it as you suggest.

Reply:
We don't have a particular style guide, but I would say generally our preference isn't usually to try to save on vertical space.

We should add this as a lint rule though